dev-java/asm/asm-5.1.ebuild

```diff
@@ -1,23 +1,26 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-OLD_P="${PN}-4.0"
+EAPI=7
+
 JAVA_PKG_IUSE="doc source"
 
 inherit java-pkg-2 java-ant-2
 
 DESCRIPTION="Bytecode manipulation framework for Java"
-HOMEPAGE="http://asm.ow2.org"
-SRC_URI="http://download.forge.objectweb.org/${PN}/${P}.tar.gz
-	http://download.forge.objectweb.org/${PN}/${OLD_P}.tar.gz"
+HOMEPAGE="https://asm.ow2.io"
+MY_P="ASM_${PV//./_}"
+SRC_URI="https://gitlab.ow2.org/asm/asm/-/archive/${MY_P}/asm-${MY_P}.tar.gz https://gitlab.ow2.org/asm/asm/-/archive/ASM_4_0/asm-ASM_4_0.tar.gz"
+
 LICENSE="BSD"
 SLOT="4"
-IUSE=""
 KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~x64-macos"
 
-DEPEND=">=virtual/jdk-1.6"
-RDEPEND=">=virtual/jre-1.6"
+CDEPEND=""
+DEPEND=">=virtual/jdk-1.8:*"
+RDEPEND=">=virtual/jre-1.8:*"
```

Comment:
idk if i noticed that before on other contributions, but generally, this should be done in revbump because it changes the bytecode of the jar files. if we do it without revbump, the package won't get rebuilt which might cause issues in future, because future jdks might not support 1.6 bytecode and hence the jar might not be loadable by the jdk and package rebuild would be the only fix, which would not be obvious though.

Reply:
Revbump done.  Thanks for clear explanation.

Reply:
I don't think the eapi bump should be a new revision because it doesn't break the package.

Reply:
Done anyway. Problems with revbump?